sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/BlobDecryptionPolicy.java

```diff
@@ -215,11 +205,10 @@
                             }
                         }
                         totalInputBytes.addAndGet(bytesToInput);
-                        plaintextByteBuffer.position(0); // Reset the position after writing.
 
                         // This is to get the decryptedBytes value into the next map to adjust the position of the
                         // output buffer because the data range was increased for decryption.
-                        plaintextByteBuffer.limit(decryptedBytes);
+                        plaintextByteBuffer.flip();
```

Comment:
I'm not confident this change preserves the behavior. DecryptedBytes is the number of bytes stored in the output buffer whereas flip sets the limit to the position of plaintextByteBuffer, which is how many bytes were read. These might not be equivalent in the cases where plaintextByteBuffer didn't align to a block boundary, so some of the data is still held in the cipher, or where we called doFinal and there was padding that was read but was stripped by the cipher and therefore not output.